ALSpringBackend/src/main/java/com/ardaslegends/data/domain/Army.java

```diff
@@ -0,0 +1,63 @@
+package com.ardaslegends.data.domain;
+
+import lombok.*;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import java.util.List;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+
+@Entity
+@Table(name = "armies")
+public class Army {
+
+    @Id
+    private String name; //unique, the army's name
```

Comment:
Why not.

Reply:
They are static attributes that are initialize and forget. Having setters it not safe.

Reply:
Army name is quite a good example of something not being static, name changes. While it is the Id of the entity, jpa handles all the remapping in the database.

Reply:
The thing is that those things should not change at all, JPA or not. A wrong use of a Set and we are good on trying to debug hours, trying to find the cause of wrong data. Maybe the name here is a bad example, but many other attributes where I have commented are static through the object's lifetime. So maybe just add a final keyword, instead of mnually adding setters for those that need.

Reply:
Though if we are to rename armies, we should have a command for that. As it doesn't I implied the army names don't change.

Reply:
Setting the ID of the instanced object does not automatically mean that it will get changed in the databse. Sometimes you want to create an exact copy of an object to do stuff (or for example if we want to update the object in the database). When doing that you need a way to set the ID of the Object manually

Reply:
If you want to create an exact copy you clone it (Prototype for reference), but in this case you do not keep it's identity (else if you wanna keep it why create a copy). If you want to update it in the database, why would you want to change the ID of something. It removes the purpose of the ID no? 

It's a very common source of bugs, hence why it's very recommended to make them final. Good practices :D

Reply:
> Though if we are to rename armies, we should have a command for that. As it doesn't I implied the army names don't change.

yeah probably, just a thought since sometimes users will ask us staff if we can rename armies (too long). Solving this with a command is prob the most elegant solution as manually changing the databse (sql updates) requires a reboot of the spring app

Reply:
So I basically looked around a bit and (i forgot that) jpa needs to be able to set the id on the calling of the repository.save(object) method. Which is why I never saw the Id field being set to final.

Reply:
Alright no problem then if it's needed. For the rest however I would prefer if they were final :D 
Small detail but is important but the general comprehension of the codebase

Reply:
https://www.objectdb.com/java/jpa/entity/fields
![image](https://user-images.githubusercontent.com/59726228/170845664-4baa7e9e-b377-4b6c-84d6-103dc2217e6d.png)